File: head first OOAD Guitar Python/head first OOAD Guitar/GuitarSpec.py
class GuitarSpec:
    def __init__(self,builder,model,type,numStrings,backWood,topWood):
        self._builder = builder
        self._model = model
        self._type = type
        self._numStrings = numStrings
        self._backWood = backWood
        self._topWood = topWood
        
    def getBuilder(self):
        return self._builder

    def getModel(self):
        return self._model

    def getType(self):
        return self._type

    def getNumStrings(self):
        return self._numStrings

    def getBackWood(self):
        return self._backWood

    def getTopWood(self):
        return self._topWood

    def matches(self,otherSpec):
        if self._builder != otherSpec._builder:
            return False
        elif self._model != None and self._model != "" and self._model.lower() != otherSpec._model.lower():
            return False
        elif self._type != otherSpec._type:
            return False
        elif self._numStrings != otherSpec._numStrings:
            return False
        elif self._backWood != otherSpec._backWood:
            return False
        elif self._topWood != otherSpec._topWood:
            return False
        else: return True

File: head first OOAD Guitar/head first OOAD Guitar/GuitarSpec.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace head_first_OOAD_Guitar
{
    class GuitarSpec
    {
        private Builder builder;
        private String model;
        private Type type;
        private int numStrings;
        private Wood backWood, topWood;
        public GuitarSpec(Builder builder, String model, Type type, int numStrings, Wood backWood, Wood topWood)
        {
            this.builder = builder;
            this.model = model;
            this.type = type;
            this.numStrings = numStrings;
            this.backWood = backWood;
            this.topWood = topWood;
        }
        public Builder getBuilder()
        {
            return builder;
        }
        public String getModel()
        {
            return model;
        }
        public Type getType()
        {
            return type;
        }
        public int getNumStrings()
        {
            return numStrings;
        }
        public Wood getBackWood()
        {
            return backWood;
        }
        public Wood getTopWood()
        {
            return topWood;
        }
        public bool matches(GuitarSpec otherSpec)
        {
            if (builder != otherSpec.builder)
                return false;
            else if ((model != null) && (!model.Equals("")) && (!model.ToLower().Equals(otherSpec.model.ToLower())))
                return false;
            else if (type != otherSpec.type)
                return false;
            else if (numStrings != otherSpec.numStrings)
                return false;
            else if (backWood != otherSpec.backWood)
                return false;
            else if (topWood != otherSpec.topWood)
                return false;
            else return true;
        }
    }
}
